### PR TITLE
PAL out NameResolution and port tests.

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

**Comment by pgavlin** [comment]
I should note that the tests currently fail due to an NYI in `Socket.InitializeSockets()`.

File: src/System.Net.NameResolution/tests/FunctionalTests/System.Net.Primitives.Functional.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (CIPop) [inline src/System.Net.NameResolution/tests/FunctionalTests/System.Net.Primitives.Functional.Tests.csproj]: @pgavlin I think you meant System.Net.NameResolution.Functional.Tests.csproj.

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj [hunk 2]
```diff
@@ -21,6 +21,8 @@
   <ItemGroup>
     <Compile Include="System\Net\DNS.cs" />
     <Compile Include="System\Net\IPHostEntry.cs" />
+    <Compile Include="System\Net\NameResolutionPal.Windows.cs" />
+    <Compile Include="System\Net\NameResolutionUtilities.cs" />
```
Comment (CIPop) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj:25]: @pgavlin Are you missing NameResolutionUtilities.cs?

**Comment by CIPop** [comment]
@pgavlin Please review above observations (especially the missing file) then, after everything builds (but fails the tests), please merge the PR.

File: src/System.Net.NameResolution/tests/FunctionalTests/System.Net.Primitives.Functional.Tests.csproj (hunk 1 above)
Comment (pgavlin) [inline src/System.Net.NameResolution/tests/FunctionalTests/System.Net.Primitives.Functional.Tests.csproj]: Indeed I did. Good catch on the copypasta.

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj (hunk 2 above)
Comment (pgavlin) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj:25]: Just fixed this.

**Pull request merged**